### PR TITLE
Device Simulation

### LINKED ISSUE #2: Device simulation for tests
- Simulating device and send values on Kafka
  - Plasmacutter
  - Filter
- Commandline tool in Rust
- Press buttons to send data for demos

### LINKED ISSUE #7: Integrate Rust static analysis/linting tool as part of the github workflow
https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/.github/workflows/quality.yaml

### PR DESCRIPTION
**Pull request by sysarcher**
Device simulator code.﻿

Todos:
- [x] Evaluate librdkafka for client
- [x] Change CLI to TOML config
- [x] Create CI/CD action
- [x] Add security scans (dependabot)
- [x] Update `README.md`
- [x] Accept config file from CLI

Closes #2 and closes #7

### CODE REVIEW

File: gateway/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+version = "0.1.0"
+authors = ["sysarcher <sysarcher@users.noreply.github.com>"]
+edition = "2018"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+pancurses = "0.16.1"
+rand = "0.8.4"
+#kafka = "0.8"
```
Comment (wagmarcel) [inline gateway/Cargo.toml:12]: if not used, please remove it or provide reason why it should stay

File: gateway/Cargo.lock [hunk 2]
```diff
@@ -0,0 +1,836 @@
+# This file is automatically @generated by Cargo.
```
Comment (wagmarcel) [inline gateway/Cargo.lock:1]: Can this be created automatically? If yes, can we autogenerate is rather than commit it

File: gateway/gateway.toml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+broker = "localhost:9094"
+topic = 'topic'
+
+[messages]
+list_of_msgs = [
+    '{"aid":"aid","cid":"cid","dataType":"String","on":"1637825686000","systemOn":"1637825686000","value":"0.005"}',
```
Comment (wagmarcel) [inline gateway/gateway.toml:6]: are the timestamps updated somewhere?

**Review by wagmarcel — CHANGES_REQUESTED** [review]

File: gateway/Cargo.lock (hunk 2 above)
Comment (sysarcher) [reply gateway/Cargo.lock:1]: @wagmarcel it's recommended to keep `Cargo.lock` file for binaries and ignore it for libs

https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

File: gateway/gateway.toml (hunk 3 above)
Comment (sysarcher) [reply gateway/gateway.toml:6]: woops, good catch!!! They aren't, I'll add that code

File: gateway/Cargo.toml (hunk 1 above)
Comment (sysarcher) [reply gateway/Cargo.toml:12]: done with the next commit

**Review by wagmarcel — APPROVED** [review]

**Pull request merged**